pkgs/development/python-modules/notify-py/default.nix

```diff
@@ -12,22 +13,27 @@ buildPythonPackage rec {
   };
 
   postPatch = ''
-   substituteInPlace setup.py \
-     --replace "loguru==0.4.1" "loguru~=0.5.0"
+    substituteInPlace setup.py \
+      --replace "loguru==0.4.1" "loguru~=0.5.0"
   '';
 
-  propagatedBuildInputs = [ alsaUtils libnotify loguru which ];
+  propagatedBuildInputs = [ loguru which ]
+    ++ lib.optionals stdenv.isLinux [ alsaUtils libnotify ];
 
-  checkInputs = [ alsaUtils libnotify pytest which ];
+  checkInputs = [ pytest which ]
+    ++ lib.optionals stdenv.isLinux [ alsaUtils libnotify ];
 
   checkPhase = ''
     pytest
   '';
 
+  # Tests search for "afplay" binary which is built in to MacOS and not available in nixpkgs
+  doCheck = stdenv.isLinux;
```

Comment:
This is the best I could come up with. Tests fail when building because MacOS' `afplay` isn't available in `nixpkgs`, but when running in `nix-shell` it actually works fine (guessing because it's an impure shell). For example, the following works on mac and plays the sound:

```sh
result/bin/notifypy --title test_title --message test_message --applicationName test_app --soundPath doorbell.wav
```

Is there a better way to deal with a binary like `afplay`?

Reply:
You could disable the one specific test that requires this binary.

Reply:
You could also try to mock it out by adding a stub script that does nothing.

Reply:
Yeah I was just wondering if there was some way to just _use_ the built-in Mac binary. I know it doesn't really go with the Nix philosophy, but for Mac that's just the way some of this stuff works.

Reply:
I think there is a way of pulling in macos frameworks. I don't know if the same works for binaries. Not a darwin user.

Reply:
Yeah I tried that as well earlier, at least with `CoreMedia` framework (just as a wild guess, no clue if any of the frameworks actually somehow make `afplay` available), no luck.

Reply:
nix-locate does not know about any afplay.

Reply:
Yeah I couldn't find any reference, either. But to be clear just asking if there's any general pattern for dealing with something _like_ `afplay`, which doesn't seem like it can be packaged but we know it will be there on the target OS.

Reply:
Possible workaround for `afplay`: https://github.com/NixOS/nixpkgs/pull/120289/files#r619559389

Reply:
Doing some more research, also found [`__impureHostDeps` and `__propagatedImpureHostDeps`](https://github.com/NixOS/nixpkgs/issues/67166). And also some sort of setting(?) for [`allowed-impure-host-deps`](https://github.com/NixOS/nix/blob/fe2bf464cf1cdbed11d4a38f063c4f8b1c922a48/src/libstore/globals.hh#L474).

Reply:
ping @NixOS/darwin-maintainers